src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.FixAllProvider.cs

```diff
@@ -26,40 +26,29 @@ public async override Task<CodeAction> GetFixAsync(FixAllContext fixAllContext)
-                        RemoveSuppressionCodeAction.GetBatchFixer(suppressionFixer);
+                if (NestedSuppressionCodeAction.IsEquivalenceKeyForGlobalSuppression(fixAllContext.CodeActionEquivalenceKey))
+                {
+                    // For global suppressions, we defer to the global suppression system to handle directly.
+                    var title = fixAllContext.CodeActionEquivalenceKey;
+                    return fixAllContext.Document != null
+                        ? GlobalSuppressMessageFixAllCodeAction.Create(
+                            title, suppressionFixer, fixAllContext.Document,
+                            await fixAllContext.GetDocumentDiagnosticsToFixAsync().ConfigureAwait(false))
+                        : GlobalSuppressMessageFixAllCodeAction.Create(
+                            title, suppressionFixer, fixAllContext.Project,
+                            await fixAllContext.GetProjectDiagnosticsToFixAsync().ConfigureAwait(false));
                 }
 
-                var title = fixAllContext.CodeActionEquivalenceKey;
-                if (fixAllContext.Document != null)
-                {
-                    var documentsAndDiagnosticsToFixMap =
-                        await fixAllContext.GetDocumentDiagnosticsToFixAsync().ConfigureAwait(false);
+                var isPragmaWarningSuppression = NestedSuppressionCodeAction.IsEquivalenceKeyForPragmaWarning(fixAllContext.CodeActionEquivalenceKey);
+                Contract.ThrowIfFalse(isPragmaWarningSuppression || NestedSuppressionCodeAction.IsEquivalenceKeyForRemoveSuppression(fixAllContext.CodeActionEquivalenceKey));
 
-                    return !isGlobalSuppression
-                        ? await batchFixer.GetFixAsync(
-                            documentsAndDiagnosticsToFixMap, fixAllContext.State, fixAllContext.CancellationToken).ConfigureAwait(false)
-                        : GlobalSuppressMessageFixAllCodeAction.Create(title, suppressionFixer, fixAllContext.Document, documentsAndDiagnosticsToFixMap);
-                }
-                else
-                {
-                    var projectsAndDiagnosticsToFixMap =
-                        await fixAllContext.GetProjectDiagnosticsToFixAsync().ConfigureAwait(false);
+                var batchFixer = isPragmaWarningSuppression
+                    ? new PragmaWarningBatchFixAllProvider(suppressionFixer)
+                    : RemoveSuppressionCodeAction.GetBatchFixer(suppressionFixer);
 
-                    return !isGlobalSuppression
-                        ? await batchFixer.GetFixAsync(
-                            projectsAndDiagnosticsToFixMap, fixAllContext.State, fixAllContext.CancellationToken).ConfigureAwait(false)
-                        : GlobalSuppressMessageFixAllCodeAction.Create(title, suppressionFixer, fixAllContext.Project, projectsAndDiagnosticsToFixMap);
-                }
+                return await batchFixer.GetFixAsync(fixAllContext).ConfigureAwait(false);
```

Comment:
this actually all got a *lot* simpler.  

First, we didn't ever even use `WellKnownFixAllProviders.BatchFixer`!!  note that in the original code either we're a global suppression (and we don't use it), or we're not global and we replace it with the pragmawarning or removesuppression batch fixers! :)

Second, we had multiple codepaths doing special stuff based on if this was global suppression or not.  Now i just have the global suppression logic in one place, and the normal logic below that.